src/TrifanHarness.cpp

```diff
@@ -68,27 +69,21 @@ int main()
         {
             printHelp();
         }
-        else if(flight_command == "takeoff")
+        else if(flight_command.substr(0, 8) == "takeoff ")
         {
             // probably should check arm/disarm here and abort if
             // disarmed
-            ctrl->updateMotors(3500);
-            // if commanded to end takeoff, set motors to equillibrium
-            std::string takeoff_command;
-            while(true)
+            // parse the takeoff altitude parameter and convert to double
+            std::regex r("\\b\\d+");
+            std::smatch match;
+            if (std::regex_search(flight_command, match, r) == true)
             {
-              std::cout << "Enter 'end' to end takeoff operation.\n";
-              std::getline(std::cin, takeoff_command);
-              if (takeoff_command == "end")
-              {
-                ctrl->updateMotors(3000);
-                std::cout << "Takeoff operation complete.\n";
-                break;
-              }
-              else
-              {
-                std::cout << "Invalid command.\n";
-              }
+              std::string alt_param_str = match.str(0);
+              double alt_param = std::stod(alt_param_str);
+              // set takeoff altitude
+              ctrl->setTakeoffAlt(alt_param);
+              // increase motor speed to gain altitude
+              ctrl->updateMotors(3500);
```

Comment:
This is where I would go into a loop (while [true] ) and check/print altitude, until you hit the target altitude, and then break out of the loop.  This does still restrict their action, but it restricts it to exactly what they specified (basically the whole thing blocks until they hit the altitude they wanted to takeoff to).  So if you remove the simulator check and put it here instead (make sure to sleep for a second on each loop to prevent from just checking every millisecond), that should make it good 👍

Reply:
okay, that makes sense. I wasn't sure if a loop within the trifan harness would be out of sync with the loop in the simulator that processes the altitude change